src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -283,15 +292,65 @@ export default function CDPipeline({
         setAvailableTags(tags)
     }
 
-    useEffect(() => {
-        getInit()
-    }, [])
+    const handlePopulatePluginDataStore = async (form: PipelineFormType) => {
+        if (!form) {
+            return
+        }
 
-    useEffect(() => {
-        if (formData.environmentId) {
-            getConfigMapSecrets()
+        const preBuildPluginIds = getPluginIdsFromBuildStage(form.preBuildStage)
+        const postBuildPluginIds = getPluginIdsFromBuildStage(form.postBuildStage)
+        const uniquePluginIds = Array.from(new Set([...preBuildPluginIds, ...postBuildPluginIds]))
+
+        if (processPluginData) {
+            await getMandatoryPluginData(form, uniquePluginIds)
+            return
+        }
+
+        if (uniquePluginIds.length === 0) {
+            return
         }
-    }, [formData.environmentId])
+
+        const {
+            pluginStore: { parentPluginStore, pluginVersionStore },
+        } = await getPluginsDetail({
+            appId: +appId,
+            pluginIds: uniquePluginIds,
+            shouldShowError: false,
+        })
+
+        handlePluginDataStoreUpdate(getUpdatedPluginStore(pluginDataStore, parentPluginStore, pluginVersionStore))
+    }
+
+    const getEnvCDPipelineName = (form) => {
+        Promise.all([getCDPipelineNameSuggestion(appId), getEnvironmentListMinPublic(true)])
```

Comment:
Why are we using the deprecated API here?

Reply:
Fixed in https://github.com/devtron-labs/dashboard/pull/2174/files